catroid/src/main/java/org/catrobat/catroid/utils/IconsUtil.java

```diff
@@ -0,0 +1,386 @@
+				} else if (category.equals(context.getString(R.string.category_arduino))) {
+					drawable = context.getResources().getDrawable(R.drawable.arduino_pos);
+				} else if (category.equals(context.getString(R.string.category_drone))) {
+					drawable = context.getResources().getDrawable(R.drawable.ardrone_pos);
+				} else if (category.equals(context.getString(R.string.category_phiro))) {
+					drawable = context.getResources().getDrawable(R.drawable.phiro_pos);
+				} else if (category.equals(context.getString(R.string.category_raspi))) {
+					drawable = context.getResources().getDrawable(R.drawable.raspberrypi_pos);
+				} else if (category.equals(context.getString(R.string.category_event))) {
+					drawable = context.getResources().getDrawable(R.drawable.event_pos);
+				} else if (category.equals(context.getString(R.string.category_pen))) {
+					drawable = context.getResources().getDrawable(R.drawable.pen_pos);
+				}
+			}
+
+			if (drawable != null) {
+				if (isLargeSize()) {
+					drawable.setBounds(largeIconSizeCategory);
+				} else {
+					drawable.setBounds(smallIconSizeCategory);
+				}
+				textView.setCompoundDrawables(drawable, null, null, null);
+			}
+		}
+	}
+
+	public static void addIcons(ViewGroup viewGroup, String category) {
+		if (isActivated()) {
+			for (int i = 0; i < viewGroup.getChildCount(); i++) {
+				if (viewGroup.getChildAt(i) instanceof BrickLayout) {
+					BrickLayout l = (BrickLayout) viewGroup.getChildAt(i);
+					if (l.getChildAt(0) instanceof TextView) {
+						TextView textView = (TextView) l.getChildAt(0);
+						addIcon(viewGroup.getContext(), textView, category);
+					} else if (l.getChildAt(0) instanceof RelativeLayout) {
+						RelativeLayout rl = (RelativeLayout) l.getChildAt(0);
+						if (rl.getChildAt(0) instanceof LinearLayout) {
+							LinearLayout ll = (LinearLayout) rl.getChildAt(0);
+							if (ll.getChildAt(0) instanceof TextView) {
+								TextView textView = (TextView) ll.getChildAt(0);
+								addIcon(viewGroup.getContext(), textView, category);
+							}
+						}
+					} else if (l.getChildAt(0) instanceof LinearLayout) {
+						LinearLayout ll = (LinearLayout) l.getChildAt(0);
+						if (ll.getChildAt(0) instanceof TableRow) {
+							TableRow tr = (TableRow) ll.getChildAt(0);
+							if (tr.getChildAt(0) instanceof TextView) {
+								TextView textView = (TextView) tr.getChildAt(0);
+								addIcon(viewGroup.getContext(), textView, category);
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	public static void setMainMenuIconSize(Context context, View contentView) {
+		if (!isLargeSize()) {
+			return;
+		}
+
+		Drawable drawable = null;
+		Button button = null;
+
+		drawable = context.getResources().getDrawable(R.drawable.ic_main_menu_continue);
+		drawable.setBounds(largeIconSizeMainMenu);
+		button = (Button) contentView.findViewById(R.id.main_menu_button_continue);
+		button.setCompoundDrawables(drawable, null, null, null);
```

Comment:
This is a lot of duplicated code. Please pass in the view id and drawable id to the method instead of duplicating all that method calls.